tinygrad/renderer/llvmir.py

```diff
@@ -63,7 +63,12 @@ class LLVMRenderer(Renderer):
     BinaryOps.MAX: lambda builder, x, y, dtype: builder.select(builder.icmp_unsigned(">", x, y) if is_bool_or_unsigned(dtype) else builder.icmp_signed(">", x, y) if dtypes.is_int(dtype) else builder.fcmp_unordered(">", x, y, flags=MFLAGS), x, y),  # noqa: E501
     BinaryOps.MOD: lambda builder, x, y, dtype: builder.urem(x, y) if is_bool_or_unsigned(dtype) else builder.srem(x, y) if dtypes.is_int(dtype) else builder.frem(x, y),  # noqa: E501
     BinaryOps.XOR: lambda builder, x, y, dtype: builder.xor(x, y), BinaryOps.AND: lambda builder, x, y, dtype: builder.and_(x, y), BinaryOps.OR: lambda builder, x, y, dtype: builder.or_(x, y), # noqa: E501
+    BinaryOps.SHL: lambda builder, x, y, dtype: builder.shl(x,y), BinaryOps.SHR: lambda builder, x, y, dtype: builder.lshr(x,y) if dtypes.is_unsigned(dtype) else builder.ashr(x,y), # noqa: E501
     TernaryOps.WHERE: lambda builder, x, y, z, dtype: builder.select(x, y, z)}
+  extra_matcher = PatternMatcher([
```

Comment:
either enforce this dtype alignment in all backends, or just cast in the `code_for_op` similar to cstyle.